### DISCUSSION TITLE
Can autogen directly execute the sql queries or we have to use python to execute sql quries..?

### DISCUSSION THREAD

**Discussion by MihirDavada** [microsoft/autogen#1558] [Q&A]
```
@code_executor.register_for_execution()
@sql_query_generator.register_for_llm(name="sql_query_generator", description="execute the query which is generated by sql_query_generator")
def query_executer(query: Annotated[str, "User input"]):
    conn = sqlite3.connect("test.db")
    cur = conn.cursor()

    try:
        cur.execute(query)
        rows = cur.fetchall()
        return rows
    finally:
        cur.close()
        conn.close()

```

**Comment by ekzhu** [comment]
Yes. This is the right way currently. One thing I would note is that any annotated type hints will be used by LLM, so you can add a return type hint to this function to help LLM understand its usage better.

**Reply by yonitjio** [reply]
Hi, I made another approach using code executor. I think it's more flexible since it does not use function calling. I posted the code here https://github.com/microsoft/autogen/issues/28.
